client/app/hosting/framework/envvar/hosting-framework-envvar.controller.js

```diff
@@ -0,0 +1,103 @@
+angular.module("App").controller(
+    "HostingFrameworkEnvvarCtrl",
+    class HostingFrameworkEnvvarCtrl {
+
+        /**
+         * @constructs HostingFrameworkEnvvarCtrl
+         * @param $scope
+         * @param $stateParams
+         * @param $timeout
+         * @param Alerter
+         * @param Hosting
+         * @param HostingFrameworkEnvvar
+         */
+        constructor ($scope, $stateParams, $timeout, Alerter, Hosting, HostingFrameworkEnvvar) {
+            this.$scope = $scope;
+            this.$stateParams = $stateParams;
+            this.$timeout = $timeout;
+            this.Alerter = Alerter;
+            this.Hosting = Hosting;
+            this.HostingFrameworkEnvvar = HostingFrameworkEnvvar;
+        }
+
+        /**
+         * Initialize HostingFrameworkEnvvarCtrl
+         */
+        $onInit () {
+            this.loading = true;
+            this.hasResult = false;
+            this.envvars = [];
+            this.maxEnvvars = 0;
+
+            this.$scope.$on(this.Hosting.events.tabFrameworkEnvvarsRefresh, () => this.getIds());
+
+            this.getIds();
+            this.loadCapabilities();
+        }
+
+        /**
+         * Load all environment variables keys from API
+         */
+        getIds () {
+            return this.HostingFrameworkEnvvar.list(this.$stateParams.productId)
+                .then((keys) => {
+                    if (keys) {
```

Comment:
You should check if keys is an array, otherwise it won't have the `sort` method : `_.isArray(keys)`